packages/angular/src/migrations/update-15-0-0/switch-to-jasmine-marbles.ts

```diff
@@ -0,0 +1,132 @@
+import type { Tree } from '@nrwl/devkit';
+import {
+  addDependenciesToPackageJson,
+  readJson,
+  visitNotIgnoredFiles,
+} from '@nrwl/devkit';
+import { extname } from 'path';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import { jasmineMarblesVersion } from '@nrwl/angular/src/utils/versions';
+
+export default function switchToJasmineMarbles(tree: Tree) {
+  const usesJasmineMarbles = replaceJasmineMarbleUsagesInFiles(tree);
+  addJasmineMarblesDevDependencyIfUsed(tree, usesJasmineMarbles);
+}
+
+function replaceJasmineMarbleUsagesInFiles(tree: Tree) {
+  let usesJasmineMarbles = false;
+
+  visitNotIgnoredFiles(tree, '/', (path) => {
```

Comment:
We could consider reducing the number of files to visit by only traversing projects importing `@nrwl/angular` doing something like:

```ts
const dirsToTraverse = Object.entries(projectGraph.dependencies)
  .filter(([, dep]) => dep.some(({ target }) => target === 'npm:@nrwl/angular'))
  .map(([projectName]) => readProjectConfiguration(tree, projectName).root);

for (const dir of dirsToTraverse) {
  visitNotIgnoredFiles(tree, dir, (path) => {
    ...
  });
}
```

Reply:
good idea, i like it